src/main/java/io/kestra/plugin/aws/AbstractConnectionInterface.java

```diff
@@ -76,4 +93,144 @@ public interface AbstractConnectionInterface {
     )
     @PluginProperty(dynamic = true)
     String getEndpointOverride();
+
+    @PluginProperty(dynamic = true)
+    default Boolean getCompatibilityMode() {
+        return false;
+    }
+
+    static String renderStringConfig(final RunContext runContext, final String config) throws IllegalVariableEvaluationException {
```

Comment:
I'm not a fan of static methods in an interface, I think it's OK for small convenient methods or sharing code for default methods but here it really look as if you have a service inside your interface whereas interface is more used for defining a contract between multiple piece of code.

I'll extract all those to a service with static methods, this is what we usually do in plugins.